marshal/marshal.go

```diff
@@ -0,0 +1,192 @@
+package marshal
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+const force_send_field_name = "ForceSendFields"
+
+// Marshal returns a JSON encoding of the given object. Included fields:
```

Comment:
Let's document the behavior for embedded struct/anonymous fields w.r.t. ForceSendFields defined in each of them.